sympy/core/function.py

```diff
@@ -3025,7 +3025,26 @@ def expand_power_base(expr, deep=True, force=False):
     >>> expand_power_base((2*y)**(1+z))
     2**(z + 1)*y**(z + 1)
     >>> ((2*y)**(1+z)).expand()
-    2*2**z*y*y**z
+    2*2**z*y**(z + 1)
+
+    The power that is unexpanded can be expanded safely when
+    ``y != 0`` or ``z + 1 != 0``, otherwise if both are 0 but
+    the power was expanded, the whole expression would
+    become 0 instead of ``2*2**z``:
```

Comment:
```suggestion
    The power that is unexpanded can be expanded safely only
    when ``y != 0``; otherwise expansion might simplify to 0 prematurely.
```